### PR TITLE
add helper to verify the correct model output file exists

### PR DESCRIPTION
**Pull request by winglian**
# Description

Adds a helper function to check for the output model when doing e2e training tests.

## Motivation and Context

It can be cumbersome to remember which file to look for when adding the check to tests, so this handles all that logic.

## How has this been tested?

update tests!

### CODE REVIEW

File: tests/e2e/utils.py [hunk 1]
```diff
@@ -81,3 +83,22 @@ def check_tensorboard(
...
+    if cfg.save_safetensors:
+        if not cfg.adapter:
+            assert (Path(temp_dir) / "model.safetensors").exists()
+        else:
+            assert (Path(temp_dir) / "adapter_model.safetensors").exists()
+    else:
+        if not cfg.adapter:
+            assert (Path(temp_dir) / "pytorch_model.bin").exists()
+        else:
+            assert (Path(temp_dir) / "adapter_model.bin").exists()
```
Comment (NanoCode012) [inline tests/e2e/utils.py]: Do we need to check for edge case where the model is large and is shared across several files? The naming would not match in this case.

Comment (winglian) [reply tests/e2e/utils.py]: not really an issue since in our tests, almost everything is a small model that doesn't shard.

**Comment by NanoCode012** [comment]
I saw that you didn't change all files in e2e like `test_dpo.py`. Even though they had `assert (Path(temp_dir) / "checkpoint-20/adapter_model.safetensors").exists()`

Should this be changed to use your new function but pass `Path(temp_dir) / "checkpoint-20"` as input path instead?

File: tests/e2e/integrations/test_cut_cross_entropy.py [hunk 2]
```diff
@@ -2,9 +2,8 @@
 Simple end-to-end test for Cut Cross Entropy integration
 """
 
-from pathlib import Path
-
 import pytest
+from e2e.utils import check_model_output_exists
```
Comment (djsaunde) [inline tests/e2e/integrations/test_cut_cross_entropy.py]: Nit: something weird is going on here with the sort order.

**Review by djsaunde — APPROVED** [review]
Just a nit.
